wazp/callbacks.py

```diff
@@ -415,6 +420,612 @@ def modify_rows_selection(
+                # Add the frame number and the ROI name to the new shapes
+                for shape in new_graph_shapes:
+                    shape["drawn_on_frame"] = frame_num
+                    shape["roi_name"] = roi_color_mapping["color2roi"][
+                        shape["line"]["color"]
+                    ]
+                # Pass the new shapes to the storage
+                roi_storage[video_name]["shapes"] += new_graph_shapes
+
+            elif re.match("shapes\[[0-9]+\].", list(graph_relayout.keys())[0]):
+                # this means that a single shape has been edited
+                # So update only that shape in storage
+                for key in graph_relayout.keys():
+                    shape_i = int(re.findall(r"\d+", key)[0])
+                    shape_attr = key.split(".")[-1]
+                    roi_storage[video_name]["shapes"][shape_i][
+                        shape_attr
+                    ] = graph_relayout[key]
+                    roi_storage[video_name]["shapes"][shape_i][
+                        "drawn_on_frame"
+                    ] = frame_num
+
+            else:
+                # this means that the graph was zoomed/panned
+                return dash.no_update
+
+        # If triggered by the load ROIs button click
+        # Load the ROIs from the metadata file
+        # CAUTION! This will overwrite any ROIs is the roi-storage
+        elif trigger == "load-rois-button.n_clicks":
+            if load_clicks > 0:
+                metadata_path = video_path_pl.with_suffix(".metadata.yaml")
+                roi_storage[video_name]["shapes"] = utils.load_rois_from_yaml(
+                    yaml_path=metadata_path
+                )
+
+        return roi_storage
+
+    @app.callback(
+        [
+            Output("frame-graph", "figure"),
+            Output("frame-status-alert", "children"),
+            Output("frame-status-alert", "color"),
+            Output("frame-status-alert", "is_open"),
+        ],
+        [
+            Input("video-select", "value"),
+            Input("frame-slider", "value"),
+            Input("roi-select", "value"),
+            Input("roi-storage", "data"),
+        ],
+        [
+            State("frame-graph", "figure"),
+            State("roi-colors-storage", "data"),
+        ],
+    )
+    def update_frame_graph(
+        video_path: str,
+        frame_num: int,
+        roi_name: str,
+        roi_storage: dict,
+        current_fig: go.Figure,
+        roi_color_mapping: dict,
+    ) -> tuple[go.Figure, str, str, bool]:
+        """
+        Update the frame graph
+
+        Parameters
+        ----------
+        video_path : str
+            Path to the video file.
+        frame_num : int
+            Frame number (which video frame to display).
+        roi_name : str
+            Name of the next ROI to be drawn.
+        roi_storage : dict
+            Dictionary storing already drawn ROI shapes.
+        current_fig : go.Figure
```

Comment:
I'd go for the full namespace...
```suggestion
        current_fig : plotly.graph_objects.Figure
```
in the docs, right? Ditto below.

Reply:
> How easy would it be (sorry!) to implement backspace deletion? Moving the RoIs around and resizing are both possible, but my naive attempts to delete (backspace and right-click) don't do anything, I have to notice the feint backspace icon on hover at the top.

I actually thought of that as well. Unfortunately, I haven't found any plotly-native way of enabling shape deletion via clicking `delete` or `backspace`. The canonical way of doing that is through the "erase" tool on the toolbar above the graph. We could design custom callbacks, but I didn't want to complicate things.

An idea would be to add a delete functionality to rows of the table, such that if you delete a row, the corresponding shape disappears. That would be possible, but would increase the complexity of the callback dance between `frame-graph`, `roi-storage`, and `roi-table`.